### PR TITLE
Updated readme and sample experience

### PR DESCRIPTION
**Pull request by daveshanley**
Refactored some old code and moved it into a package ready to be cleaned up.

Signed-off-by: Dave Shanley <dshanley@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,113 +113,151 @@ This will output:
...
+
+	// get a pointer to the channel manager
+	cm := b.GetChannelManager()
+
+	channel := "my-stream"
+	cm.CreateChannel(channel)
+
+	// listen to stream of messages coming in on channel, a handler is returned that allows you to add in
+	// lambdas that handle your
+	handler, err := b.ListenStream(channel)
```
Comment (jooskim) [inline README.md:156]: just a suggestion, but maybe we can pull this code closer to where handler is actually defined, like L191 so the flow is smoother to readers?

File: README.md [hunk 2]
```diff
@@ -131,113 +113,151 @@ This will output:
...
+
+	// keep listening
+	handler.Handle(
+		func(msg *model.Message) {
+
+			// unmarshal the message payload into a model.Response object
+			// this is a wrapper transport uses when being used as a server, it encapsulates a rich set of data
+			// about the message, but you only really care about the payload (body)
+			r := &model.Response{}
+			d := msg.Payload.([]byte)
```
Comment (jooskim) [inline README.md:200]: i think we can just go straight to `msg.Payload.(*model.Response)` but seems like you wanted more explanation here breaking it down to the smallest bits. is that the case?

File: README.md [hunk 3]
```diff
@@ -131,113 +113,151 @@ This will output:
...
+
+	// create a broker connector config and connect to transport-bus.io over WebSocket using TLS.
+	config := &bridge.BrokerConnectorConfig{
+		Username:   "guest",            // not required for demo, but our API requires it.
+		Password:   "guest",            // ^^ same.
+		ServerAddr: "transport-bus.io", // our live broker running plank and demo services.
+		UseWS:      true,               // connect over websockets
+		WebSocketConfig: &bridge.WebSocketConfig{ // configure websocket
+			WSPath: "/ws", // websocket endpoint
+			UseTLS: true,  // use TLS/HTTPS
```
Comment (jooskim) [inline README.md:170]: i see that you embedded a default object for `TLSConfig` into the library, but maybe in the spirit of making it clear should we leave a comment here that TLSConfig will be auto-populated and also that you can supply your own TLSConfig here?

Comment (daveshanley) [reply README.md:170]: Good idea.

File: README.md (hunk 2 above)
Comment (jooskim) [reply README.md:200]: oh, it's because we're sending an encoded blob slice, yeah let's leave it like that.

File: README.md (hunk 1 above)
Comment (daveshanley) [reply README.md]: good idea. done

**Review by jooskim — APPROVED** [review]
LGTM :shipit:

**Pull request merged**